zilliqa/src/block_store.rs

```diff
@@ -49,13 +50,32 @@ pub struct BlockStore {
     /// handle pending requests made by ourselves that arrive out-of-order, while also giving some extra space for
     /// newly created blocks that arrive while we are syncing.
     buffered: LruCache<Hash, Proposal>,
+    /// Requests we would like to sent, but haven't been able to (e.g. because we have no peers).
+    pending_requests: Vec<(u64, u64)>,
```

Comment:
You're using the same name (pending) for requests we've sent to a peer (in PeerInfo) and requests you couldn't send. Perhaps change one of them to avoid confusion? unserviceable_requests?